ophyd/areadetector/filestore_mixins.py

```diff
@@ -241,25 +292,66 @@ def write_path_template(self, val):
         self._write_path_template = _ensure_trailing_slash(val)
 
     def stage(self):
-        self._point_counter = count()
         self._locked_key_list = False
         self._datum_uids.clear()
         super().stage()
-        if self.enable.get() and self._reg is None:
-            raise RuntimeError('The plugin {!r} is enabled, but '
-                               'the Registry (self._reg) is `None`')
+        # Subclasses will assemble resource_kwargs and call
+        # self._generate_resource(resource_kwrags)
+
+    def _generate_resource(self, resource_kwargs):
```

Comment:
why an underscore (compared to `generate_datum`)?

Reply:
generate_datum is part of the public API and the interface that AD uses to find it's children than emit AssetRegistry documents, this is a helper method to centralize some code.

If we could do it again I would call `generate_datum` something like `notify_children_AD_emitted_an_array_that_wrote_to_disk` (but that is a bit wordy....)